features/step_definitions/crd.rb

```diff
@@ -0,0 +1,24 @@
+Given /^the #{QUOTED} #{QUOTED} CRD is recreated after scenario$/ do |name, crd|
+  ensure_admin_tagged
+  allowed_crds = ["oauth", "kubeapiserver"]
+
+  if allowed_crds.include? crd
+    @result = admin.cli_exec(:get, resource: crd, resource_name: name, o: 'yaml')
+    if @result[:success]
+      origin_crd = @result[:response]
+      logger.info "#{crd} restore tear_down registered:\n#{origin_crd}"
+    else
+      raise "Could not get #{crd}: #{name}"
+    end
+
+    _admin = admin
+    teardown_add {
+      admin.cli_exec(:delete, object_type: crd, object_name_or_id: name)
+
+      @result = _admin.cli_exec(:create, f: "-", _stdin: origin_crd)
+      raise "Cannot restore #{crd}: #{name}" unless @result[:success]
+    }
```

Comment:
can we, instead of the step body, delegate to `hidden recreate cluster resource( after scenario)`?
```
  cb.cluster_resource_to_recreate = self.send(crd, name)
  step 'hidden recreate cluster resource after scenario'
```

Reply:
This assumes we create placeholder objects for the CRDs in question but it should be very easy. See example `lib/openshift/identity.rb`.